.github/workflows/gh-pages.yml

```diff
@@ -28,7 +28,13 @@ jobs:
           python -m pip install --upgrade pip
           python -m pip install -r requirements.txt -r requirements-dev.txt
           python -m pip install '.[all]'
+      - uses: actions/cache@v4
+        id: cache-docs
+        with:
+          path: docs/_build
+          key: ${{ runner.os }}-sphinx
       - name: Generate the documentation
+        if: steps.${{ runner.os }}-sphinx.outputs.cache-hit != 'true'
         run: |
           cd docs/
           make html
```

Comment:
@nelimee Do you mind taking a quick look at the changes? [The cache was added successfully](https://github.com/tqec/tqec/actions/caches), but I can't seem to make sphinx use the cache. The latest build is about to [take > 2 hours](https://github.com/tqec/tqec/actions/runs/15452845946/job/43498873970?pr=592).

Reply:
From what I see in the CI action, you are caching the `docs` folder, so the documentation will never be updated and will only be retrieved from cache. So yes, the docs should take longer to build. Probably around 3 to 5 minutes with a database.

What you should do (in my opinion) is the following:

1. Use a database with a known path (setting `database_path` when generating quantum circuits),
2. Cache **only** the database with this known path.
3. You will likely need [restore/save actions](https://github.com/actions/cache?tab=readme-ov-file#using-a-combination-of-restore-and-save-actions) rather than a single cache action.

Reply:
Yes, but you should not have to dig that much. Just setting `database_path` in 
https://github.com/tqec/tqec/blob/d6f90192269b96651a62f5957b193308ff1bc2c5/src/tqec/compile/graph.py#L453-L461
should be enough to generate the database there.

Reply:
`database_path` is a path to a file that can have any extension (but I prefer `.pkl` as it is saved using `pickle`) and that is used at circuit generation time, **not** at simulation time. 

There are two steps there:

1. Generate the quantum circuit using `tqec`, that's where the `database_path` should matter,
2. Simulate the resulting `.stim` quantum circuit using `stim`, that's where the `save_resume_filepath` should matter.

In theory, `save_resume_filepath` saves results along with a hash of the quantum circuit that has been used to generate those results, so that is strange that both points are plotted. I am suspecting that it is rather an issue with how we plot, I will have a quick look.

Reply:
I forgot how `start_simulation_using_sinter` was working, and in particular that it was taking a `BlockGraph` in. Then yes, you are right, the database_path argument should be added.

Reply:
I cannot replicate that locally: I get an error from `sinter` when trying to execute the cell. Do you have it too?